docs/frontend/forms.md

```diff
@@ -0,0 +1,326 @@
+                        "required": "required"
+                      }
+                    },
+                    "validators": [{ "identifier": "NotEmpty" }],
+                    "name": "tx_form_formframework[subject]"
+                  },
+                  
+                ]
+              },
+            ]
+```
+
+You can noticed that `name` and `subject` fields are nested in `Fieldset-2` which is nested in `Fieldset-1`. It means we have to render component fields in recursion. 
+For this list rendering is responsible [`T3FormFieldList` component](https://github.com/TYPO3-Initiatives/nuxt-typo3/blob/master/lib/components/T3Form/T3FormFieldList/T3FormFieldList.vue). 
+
+This part of the template is responsible for matching frontend component with type or `field.identifier`. 
+
+```vue
+<component
+  :is="getComponentField(field)"
+  v-model="model[field.name]"
+  :field="field"
+/>
+```
+
+
+For example: to display and render `input type="hidden"` we had to register `T3FormFieldHidden`.
+
+If you want to add custom textarea field with type "Textarea" then you have to register `T3FormFieldTextarea`
+
+You can also register custom field component for specific field - `T3FormFieldName` if your `field.identifier === 'name'`
+
+Default one is `T3FormField`
+
+## Add new field type
+
+At this moment we support
+
+1. regular input fields like text, email, number and single
+2. single select
+3. fieldset
+4. honeypot == hidden
+
+But you can easliy add new field type.
+
+For example we can add `FormFieldCheckbox`. In that case add new global component `components/T3FormFieldCheckbox.vue`:
+```vue
+<template>
+  <T3FormField v-bind="$props">
+    <template #default="{state}">
+      <input
+        :id="field.identifier"
+        v-model="innerValue"
+        type="checkbox"
+        :label="field.label"
+        :name="field.name || field.identifier"
+        :class="state.classes"
+        :required="required"
+        :true-value="true"
+        false-value=""
+        @input="(event) => $emit('input', event.target.value)"
+      >
+    </template>
+  </T3FormField>
+</template>
+<script>
+import { T3FormField } from '~typo3/components/T3Form'
+export default {
+  name: 'T3FormFieldCheckbox',
+  components: {
+    T3FormField
+  },
+  extends: T3FormField
+}
+</script>
+```
+
+:::tip
+Notice that we have used default `T3FormField` as the base template for new form field type. Thanks to that we can create multiple form types with the same template. On the other hand you can put all input types in one FormField component and match them by v-if.
+:::
+
+
+## Add new validation rule
+
+For model validation we use `vee-validate`. 
+By default we have setup following rules:
+
+1. required as **NotEmpty**
+2. email as **EmailAddress**
+3. regex as **RegularExpression**
+
+
+All available rules you can find [here](https://vee-validate.logaretm.com/v3/guide/rules.html#rules), but remember to use the same rules name as in TYPO3.
```

Comment:
All additional rules are available [here]. Remember to use the same rules name as in TYPO3.